internal/provider/protonvpn/updater/iptoserver.go

```diff
@@ -10,7 +10,7 @@ import (
 type ipToServer map[string]models.Server
 
 func (its ipToServer) add(country, region, city, name, hostname string,
-	free bool, entryIP netip.Addr) {
+	free bool, entryIP netip.Addr, features *models.Features) {
```

Comment:
nit just return the whole bunch of features after `free bool`, or define an unexported `features` struct in this package, up to you 😉 (and then add the `free bool` to it too.

Reply:
Will wait with this to see what you decide about the features struct in models/server.go

Reply:
The features are defined at logical server level, "free" at physical server level, didn't feel right to put it both in the same struct, but will do if you prefer it.

Reply:
This makes sense, especially since free is more of a subscription tier filter as you mentioned 😉